protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1458,182 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderTypeAndFormat(headerType string, format string) error {
+	// The type of the object. The value MUST be one of "string", "number", "integer", "boolean", or "array"
+	// See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#headerObject
+	// Note: currently not implementing array as we are only implementing this in the operation response context
+	switch headerType {
+	// the format property is an open string-valued property, and can have any value to support documentation needs
+	// primary check for format is to ensure that the number/integer formats are extensions of the specified type
+	// See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#dataTypeFormat
+	case "string":
+		return nil
+	case "number":
+		if format != "" {
+			switch format {
+			case
+				"uint",
+				"uint8",
+				"uint16",
+				"uint32",
+				"uint64",
+				"int",
+				"int8",
+				"int16",
+				"int32",
+				"int64",
+				"float",
+				"float32",
+				"float64",
+				"complex64",
+				"complex128",
+				"double",
+				"byte",
+				"rune",
+				"uintptr":
+				return nil
+			}
+			return fmt.Errorf("the provided format %q is not a valid extension of the type %q", format, headerType)
+		}
+		return nil
+	case "integer":
+		if format != "" {
+			switch format {
+			case
+				"uint",
+				"uint8",
+				"uint16",
+				"uint32",
+				"uint64",
+				"int",
+				"int8",
+				"int16",
+				"int32",
+				"int64":
+				return nil
+			}
+			return fmt.Errorf("the provided format %q is not a valid extension of the type %q", format, headerType)
+		}
+		return nil
+	case "boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueTypeAndFormat(headerType string, defaultValue string, format string) error {
+	switch headerType {
+	case "string":
+		if !isQuotedString(defaultValue) {
+			return fmt.Errorf("the provided default value %q does not match provider type %q, or is not properly quoted with escaped quotations", defaultValue, headerType)
+		}
+		if format != "" {
+			switch format {
+			case "date-time":
+				unquoteTime := defaultValue[1 : len(defaultValue)-1]
```

Comment:
I think we can use the string utilities here to avoid crashing on malformed strings:
```suggestion
				unquoteTime := strings.Trim(defaultValue, `"`).
```